instrumentation/kamon-lagom/src/main/scala/kamon/instrumentation/lagom/LagomMetrics.scala

```diff
@@ -0,0 +1,49 @@
+package kamon.instrumentation.lagom
+
+import kamon.Kamon
+import kamon.metric.{Counter, Gauge, Histogram, InstrumentGroup, MeasurementUnit, Metric, Timer}
+import kamon.tag.TagSet
+
+object LagomMetrics {
+
+  val defaultTags: TagSet = TagSet.of("component", "lagom")
+
+  val CBLatency: Metric.Histogram = Kamon.histogram(
+    name = "lagom.cb.latency",
+    description = "Latency of Lagom Circuit Breakers",
+    unit = MeasurementUnit.time.nanoseconds
+  )
+
+  val CBState: Metric.Gauge = Kamon.gauge(
+    name = "lagom.cb.state",
+    description = "State of Lagom Circuit Breakers (possible values: 1 - Open, 2 - HalfOpen, 3 - Closed)"
+  )
+
+  val CBCounter: Metric.Counter = Kamon.counter(
+    name = "lagom.cb.count",
+    description = "Count of calls of Lagom Circuit Breakers"
+  )
+
+  val CBTimer: Metric.Timer = Kamon.timer(
+    name = "lagom.cb.timer",
+    description = "Timer of calls of Lagom Circuit Breakers"
+  )
```

Comment:
I'm not really sold on the idea of having these three separate metrics:
- `lagom.cb.latency`
- `lagom.cb.count`
- `lagom.cb.timer`

To me, those three metrics look like they should be a single timer metric called `lagom.cb.call.duration` with these tags:
- `lagom.cb.name` with the name of the circuit breaker
- `status_code` with possible values `Ok` and `Error`
- `exception.type` with possible values `CircuitBreakerOpenException` or the one that gets triggered with timeouts. This tag would only be present when the timer was incremented on the `onCallTimeoutFailure` and `onCallBreakerOpenFailure` callbacks

That single timer can be filtered and grouped by tags to get any data you need out of it. And timers always export their "count" series too, that would make the counter unnecessary as well.

We will soon move a few things to follow the OpenTelemetry semantic conventions, so I'm trying to adapt these names to the [status values](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status) (we will end up doing something very similar in our Span metrics) and [exception](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md#attributes) conventions. I'm not super sure about the tag names so any other suggestions would be appreciated :smile:

Reply:
@ivantopo Sorry for the long silence, Ivan 😞 
I'm back and change code by your comments 😉 
Now the result looks next:

```
# HELP lagom_cb_state State of Lagom Circuit Breakers (possible values: 1 - Open, 2 - HalfOpen, 3 - Closed)
# TYPE lagom_cb_state gauge
lagom_cb_state{cb_name="circuit-breaker-name",component="lagom"} 3.0
# HELP lagom_cb_call_duration_seconds Call duration timer of Lagom Circuit Breakers
# TYPE lagom_cb_call_duration_seconds histogram
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="0.005",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="0.01",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="0.025",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="0.05",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="0.075",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="0.1",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="0.25",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="0.5",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="0.75",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="1.0",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="2.5",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="5.0",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="7.5",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="10.0",component="lagom"} 11.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",le="+Inf",component="lagom"} 11.0
lagom_cb_call_duration_seconds_count{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",component="lagom"} 11.0
lagom_cb_call_duration_seconds_sum{status_code="Error",exception_type="TimeoutException",cb_name="circuit-breaker-name",component="lagom"} 109.991428096
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="0.005",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="0.01",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="0.025",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="0.05",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="0.075",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="0.1",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="0.25",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="0.5",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="0.75",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="1.0",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="2.5",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="5.0",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="7.5",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="10.0",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",le="+Inf",component="lagom"} 6.0
lagom_cb_call_duration_seconds_count{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",component="lagom"} 6.0
lagom_cb_call_duration_seconds_sum{status_code="Error",exception_type="CircuitBreakerOpenException",cb_name="circuit-breaker-name",component="lagom"} 0.000000006
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="0.005",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="0.01",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="0.025",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="0.05",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="0.075",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="0.1",component="lagom"} 0.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="0.25",component="lagom"} 1.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="0.5",component="lagom"} 2.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="0.75",component="lagom"} 2.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="1.0",component="lagom"} 3.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="2.5",component="lagom"} 3.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="5.0",component="lagom"} 3.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="7.5",component="lagom"} 3.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="10.0",component="lagom"} 3.0
lagom_cb_call_duration_seconds_bucket{status_code="Ok",cb_name="circuit-breaker-name",le="+Inf",component="lagom"} 3.0
lagom_cb_call_duration_seconds_count{cb_name="circuit-breaker-name",status_code="Ok",component="lagom"} 3.0
lagom_cb_call_duration_seconds_sum{cb_name="circuit-breaker-name",status_code="Ok",component="lagom"} 1.485832192
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="0.005",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="0.01",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="0.025",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="0.05",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="0.075",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="0.1",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="0.25",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="0.5",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="0.75",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="1.0",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="2.5",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="5.0",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="7.5",component="lagom"} 6.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="10.0",component="lagom"} 17.0
lagom_cb_call_duration_seconds_bucket{status_code="Error",cb_name="circuit-breaker-name",le="+Inf",component="lagom"} 17.0
lagom_cb_call_duration_seconds_count{cb_name="circuit-breaker-name",status_code="Error",component="lagom"} 17.0
lagom_cb_call_duration_seconds_sum{cb_name="circuit-breaker-name",status_code="Error",component="lagom"} 109.991428102```

Reply:
I think this is ready to go now!